lib/token_helpers.ex

```diff
@@ -0,0 +1,29 @@
+defmodule TokenHelpers do
+  import QueryAdapters
+
+  def _slug_input(message_prompt) do
+    input = IO.gets(message_prompt)
+    if input == :eof do
+      raise "End of File! Check for propper mocks and imports"
+    else
+      String.trim(input)
+    end
+  end
```

Comment:
`_slug_input/1` is defined in two different modules – I know we talked about not reaching too early for an abstraction, but it feels like it could be pulled out into an IO-specific module, if you're using it in a few places? Also, is it even possible to get an `:eof`? When I enter an empty string, it doesn't give me that error or anything.

Reply:
The :eof error happens in testing when _slug_input isn't properly mocked. I'm not exactly sure what causes the error but I think the test framework assumes the process is done and spits out the :eof call.

Reply:
slug_input has been moved to IoFunctions